addon/components/ember-table.js

```diff
@@ -0,0 +1,564 @@
+
+  // Sets which row selection behavior to follow. Possible values are 'none'
+  // (clicking on a row does nothing), 'single' (clicking on a row selects it
+  // and deselects other rows), and 'multiple' (multiple rows can be selected
+  // through ctrl/cmd-click or shift-click).
+  selectionMode: 'single',
+
+  // ---------------------------------------------------------------------------
+  // API - Outputs
+  // ---------------------------------------------------------------------------
+
+  // An array of the rows currently selected. If `selectionMode` is set to
+  // 'single', the array will contain either one or zero elements.
+  selection: Ember.computed(function(key, val) {
+    var selectionMode = this.get('selectionMode');
+    if (arguments.length > 1 && val) {
+      this.get('persistedSelection').clear();
+      this.get('rangeSelection').clear();
+      switch (selectionMode) {
+        case 'single':
+          this.get('persistedSelection').addObject(val);
+          break;
+        case 'multiple':
+          this.get('persistedSelection').addObjects(val);
+      }
+    }
+    var selection = this.get('persistedSelection').copy().addObjects(this.get('rangeSelection'));
+    switch (selectionMode) {
+      case 'none':
+        return null;
+      case 'single':
+        return selection[0] || null;
+      case 'multiple':
+        return selection;
+    }
+  }).property('persistedSelection.[]', 'rangeSelection.[]', 'selectionMode'),
+
+  // ---------------------------------------------------------------------------
+  // Internal properties
+  // ---------------------------------------------------------------------------
+
+  // Special flag used by child views to look up this component using
+  // nearestWithProperty()
+  isEmberTable: true,
+
+  columnsFillTable: true,
+
+  init: function() {
+    this._super();
+    if (!Ember.$.ui) {
+      throw 'Missing dependency: jquery-ui';
+    }
+    if (!Ember.$().mousewheel) {
+      throw 'Missing dependency: jquery-mousewheel';
+    }
+    if (!Ember.$().antiscroll) {
+      throw 'Missing dependency: antiscroll.js';
+    }
+    return this.prepareTableColumns();
+  },
+
+  // TODO(azirbel): Document
+  actions: {
+    addColumn: Ember.K,
+    sortByColumn: Ember.K
+  },
+
+  height: Ember.computed.alias('_tablesContainerHeight'),
+
+  // TODO(new-api): eliminate view alias
+  // specify the view class to use for rendering the table rows
+  tableRowView: 'table-row',
+  tableRowViewClass: Ember.computed.alias('tableRowView'),
+
+  onColumnSort: function(column, newIndex) {
+    // Fixed columns are not affected by column reordering
+    var numFixedColumns = this.get('fixedColumns.length');
+    var columns = this.get('columns');
+    columns.removeObject(column);
+    columns.insertAt(numFixedColumns + newIndex, column);
+    return this.prepareTableColumns();
+  },
+
+  // An array of Ember.Table.Row computed based on `content`
+  bodyContent: Ember.computed(function() {
```

Comment:
This is probably out of scope, but we should not need to recreate the controller each time the content changes

```js
bodyContent: Ember.computed(function() {
    return RowArrayController.create({
      target: this,
      parentController: this,
      container: this.get('container'),
      itemController: Row,
      content: Ember.computed.alias 'parentController.content'
    });
  }).property()
```